### PR TITLE
REDCORE-574 Codeception for OAuth Clients

### PR DESCRIPTION
**Pull request by ThuNguyen1411**

### CODE REVIEW

File: tests/_support/Step/Acceptance/OAuthClientsStep.php [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	 * @param $clientID2
+	 * @throws \Exception
+	 */
+	public function editOAuthClient($clientID,$clientID2)
+	{
+		$I = $this;
+		$I->amOnPage(OAuthClientsPage::$URL);
+		$I->waitForText(OAuthClientsPage::$titleOAuth, 30, OAuthClientsPage::$h1);
+		$I->searchOAuthClient($clientID);
+		$I->see($clientID, OAuthClientsPage::$oauthClientsList);
```
Comment (NguyenTrang091296) [inline tests/_support/Step/Acceptance/OAuthClientsStep.php]: Please change see to waitForText

Comment (puneet0191) [reply tests/_support/Step/Acceptance/OAuthClientsStep.php]: And why should that be done?, You can add another waitForText..but it should not replace `see`, one is an implicit wait while the other is an assertion and that's what the test is supposed to do.

Comment (puneet0191) [reply tests/_support/Step/Acceptance/OAuthClientsStep.php]:
```
I->search
I->waitForText
I->see
```

Comment (NguyenTrang091296) [reply tests/_support/Step/Acceptance/OAuthClientsStep.php]: I think we should be waitForText before when use $I->see to avoid fail random

Comment (puneet0191) [reply tests/_support/Step/Acceptance/OAuthClientsStep.php]: I think that is what I mentioned here? https://github.com/redCOMPONENT-COM/redCORE/pull/868#discussion_r284191002

Comment (NguyenTrang091296) [reply tests/_support/Step/Acceptance/OAuthClientsStep.php]: Yes @puneet0191. Thanks

**Comment by nhungredweb** [comment]
@ThuNguyen1411  please help resolve conflict . thanks

File: tests/_support/Step/Acceptance/OAuthClientsStep.php [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+ */
+class OAuthClientsStep extends AbstractStep
+{
+	/**
+	 * @param $clientID
+	 * @param $redirectURI
+	 * @throws \Exception
+	 * @since 1.10.7
+	 */
+	public function createNewOAuthClient($clientID,$redirectURI)
```
Comment (NguyenTrang091296) [inline tests/_support/Step/Acceptance/OAuthClientsStep.php]: add space after ','

**Pull request merged**